lib/pages/chat/chat_app_bar_title.dart

```diff
@@ -94,71 +96,121 @@ class ChatAppBarTitle extends StatelessWidget {
                   overflow: TextOverflow.ellipsis,
                   style: ChatAppBarTitleStyle.appBarTitleStyle(context),
                 ),
-                _buildStatusContent(context, room!),
+                _ChatAppBarStatusContent(
+                  getStreamInstance: getStreamInstance,
+                  context: context,
+                  room: room!,
+                ),
               ],
             ),
           ),
         ],
       ),
     );
   }
+}
 
-  StreamBuilder<ConnectivityResult> _buildStatusContent(
-    BuildContext context,
-    Room room,
-  ) {
-    final TextStyle? statusTextStyle =
-        ChatAppBarTitleStyle.statusTextStyle(context);
+class _ChatAppBarStatusContent extends StatelessWidget {
+  const _ChatAppBarStatusContent({
+    required this.getStreamInstance,
+    required this.context,
+    required this.room,
+  });
+
+  final Stream<ConnectivityResult> getStreamInstance;
+  final BuildContext context;
+  final Room room;
 
-    return StreamBuilder<ConnectivityResult>(
-      stream: getStreamInstance,
+  @override
+  Widget build(BuildContext context) {
+    final connectivityResultStream = getStreamInstance;
+    final directChatPresenceStream = room.directChatPresenceStream;
+
+    return StreamBuilder<List>(
+      stream: StreamZip([connectivityResultStream, directChatPresenceStream])
+          .asBroadcastStream(),
       builder: (context, snapshot) {
-        if (snapshot.hasData && snapshot.data == ConnectivityResult.none) {
-          return Text(
-            L10n.of(context)!.noConnection,
-            maxLines: 1,
-            overflow: TextOverflow.ellipsis,
-            style: statusTextStyle,
-          );
+        final connectivityResult =
+            tryCast(snapshot.data?[0], fallback: ConnectivityResult.none);
+        final directChatPresence = tryCast(snapshot.data?[1], fallback: null);
```

Comment:
in the case snapshot.data length is 1, this will fall, because it both != null and need data at least length 2

Reply:
am not sure snapshot data length can be 1 since I put 2 streams https://pub.dev/documentation/async/latest/async/StreamZip-class.html